src/parser/tptp/tptp.cpp

```diff
@@ -183,36 +184,33 @@ void Tptp::includeFile(std::string fileName) {
   }
 }
 
-void Tptp::checkLetBinding(std::vector<Expr>& bvlist, Expr lhs, Expr rhs, bool formula) {
-  if(lhs.getKind() != CVC4::kind::APPLY_UF) {
+void Tptp::checkLetBinding(const std::vector<Expr>& bvlist, Expr lhs, Expr rhs,
+                           bool formula) {
+  if (lhs.getKind() != CVC4::kind::APPLY_UF) {
     parseError("malformed let: LHS must be a flat function application");
   }
-  std::vector<CVC4::Expr> v = lhs.getChildren();
+  const std::multiset<CVC4::Expr> vars{lhs.begin(), lhs.end()};
   if(formula && !lhs.getType().isBoolean()) {
     parseError("malformed let: LHS must be formula");
   }
-  for(size_t i = 0; i < v.size(); ++i) {
-    if(v[i].hasOperator()) {
-      parseError("malformed let: LHS must be flat, illegal child: " + v[i].toString());
+  for (const CVC4::Expr& var : vars) {
+    if (var.hasOperator()) {
+      parseError("malformed let: LHS must be flat, illegal child: " +
+                 var.toString());
     }
   }
-  std::sort(v.begin(), v.end());
-  std::sort(bvlist.begin(), bvlist.end());
+
   // ensure all let-bound variables appear on the LHS, and appear only once
-  for (size_t i = 0; i < bvlist.size(); ++i) {
-    std::vector<CVC4::Expr>::const_iterator found =
-        std::lower_bound(v.begin(), v.end(), bvlist[i]);
-    if (found == v.end() || *found != bvlist[i]) {
+  for (const Expr& bound_var : bvlist) {
+    const size_t count = vars.count(bound_var);
+    if (count == 0) {
       parseError(
           "malformed let: LHS must make use of all quantified variables, "
           "missing `" +
-          bvlist[i].toString() + "'");
-    }
-    assert(found != v.end());
-    std::vector<CVC4::Expr>::const_iterator found2 = found + 1;
-    if (found2 != v.end() && *found2 == *found) {
+          bound_var.toString() + "'");
+    } else if (count >= 2) {
```

Comment:
Andy can I check a stronger condition that vars does not contain any duplicates instead?

Reply:
I'm not sure, but if you think the code behaves the same way as before, I'm fine with the change.

Reply:
The proposed change would not be equivalent. The current code excludes a bvlist member from appearing twice in lhs. I am hoping to strengthen this to excluding any member of lhs from appearing twice (not just the subset in bvlist). This would allow me to simplify the code quite a bit.